arch/x86/traps.c

```diff
@@ -245,10 +246,21 @@ static char *x86_ex_decode_error_code(char *buf, size_t size, uint32_t vector,
     return buf;
 }
 
+void print_callstack(const void *sp, const void *ip) {
+    unsigned long *_sp = (unsigned long *) (_ul(sp) & ~0x7UL);
+
+    printk("CALLSTACK:\n");
+    print_symbol(ip);
+    for (int i = 0; _ul(&_sp[i]) % PAGE_SIZE_2M; i++)
+        print_symbol(_ptr(_sp[i]));
```

Comment:
Thanks for the explanation!

Reply:
Hmm, let me think how to make this better and more explicit.